pkg/util/procfs/procfs.go

```diff
@@ -45,7 +46,15 @@ func containerNameFromProcCgroup(content string) (string, error) {
 // Eg. If the devices cgroup for the container is stored in /sys/fs/cgroup/devices/docker/nginx,
 // return docker/nginx. Assumes that the process is part of exactly one cgroup hierarchy.
 func (pfs *ProcFs) GetFullContainerName(pid int) (string, error) {
-	filePath := path.Join("/proc", strconv.Itoa(pid), "cgroup")
+	var filePath string
+	// if kubelet was started with host's rootfs mounted, assume it's running as a container
+	_, err := os.Stat("/rootfs/proc")
```

Comment:
@vishh Maybe we should just inject the path and drive it from outside based
on whether kubelet is containerized, that way the 'containerized' logic
doesn't creep into ProcPs

On Tue, Sep 8, 2015 at 4:20 PM, Vish Kannan <notifications@github.com>
wrote:

> In pkg/util/procfs/procfs.go
> <https://github.com/kubernetes/kubernetes/pull/13663#discussion_r38974696>
> :
>
> > @@ -45,7 +46,15 @@ func containerNameFromProcCgroup(content string) (string, error) {
> >  // Eg. If the devices cgroup for the container is stored in /sys/fs/cgroup/devices/docker/nginx,
> >  // return docker/nginx. Assumes that the process is part of exactly one cgroup hierarchy.
> >  func (pfs *ProcFs) GetFullContainerName(pid int) (string, error) {
> > -	filePath := path.Join("/proc", strconv.Itoa(pid), "cgroup")
> > +	var filePath string
> > +	// if kubelet was started with host's rootfs mounted, assume it's running as a container
> > +	_, err := os.Stat("/rootfs/proc")
>
> @pmorie <https://github.com/pmorie> This path could be made configurable
> too. WDYT?
> +1 for injecting a containerized option.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13663/files#r38974696>.
>